MHVLib/Clients/MHVRemoteMonitoringClient.m

```diff
@@ -18,61 +18,109 @@
 //
 
 #import <Foundation/Foundation.h>
+#import "MHVCommon.h"
 #import "MHVRemoteMonitoringClient.h"
 #import "MHVClient.h"
 #import "MHVJsonSerializer.h"
+#import "MHVConnectionProtocol.h"
+#import "MHVRestRequest.h"
+#import "MHVServiceResponse.h"
+#import "MHVConnectionFactory.h"
+#import "MHVConnection.h"
+
+@interface MHVRemoteMonitoringClient ()
+
+@property (nonatomic, weak) id<MHVConnectionProtocol>     connection;
+
+@end
 
 @implementation MHVRemoteMonitoringClient
 
-+ (NSURLSessionTask*) requestWithPath:(NSString* _Nonnull)path
-                               method:(NSString* _Nonnull)method
-                           pathParams:(NSDictionary * _Nullable)pathParams
-                          queryParams:(NSDictionary* _Nullable)queryParams
-                           formParams:(NSDictionary * _Nullable)formParams
-                                 body:(id _Nullable)body
+- (instancetype)initWithConnection:(id<MHVConnectionProtocol>)connection
+{
+    MHVASSERT_PARAMETER(connection);
+    
+    self = [super init];
+    if (self)
+    {
+        _connection = connection;
+    }
+    
+    return self;
+}
+
+//PENDING: Swagger generated code needs to update to be injected with RemoteMonitoringClient & remove this method
++ (NSURLSessionTask *)requestWithPath:(NSString *_Nonnull)path
+                               method:(NSString *_Nonnull)method
```

Comment:
This one Mike will have to change in the swagger templates, then can update here